python/GafferSceneUI/RenderPassEditor.py

```diff
@@ -0,0 +1,750 @@
+	# `inspectorFunction` is a callable object of the form
+	# `inspectorFunction( scene, editScope )` returning a
+	# `GafferSceneUI._RenderPassEditor._RenderPassEditorInspectorColumn` object.
+	@classmethod
+	def registerColumn( cls, groupKey, columnKey, inspectorFunction, section = None ) :
+
+		sections = cls.__columnRegistry.setdefault( groupKey, collections.OrderedDict() )
+		section = sections.setdefault( section, collections.OrderedDict() )
+
+		section[columnKey] = inspectorFunction
+
+	def __repr__( self ) :
+
+		return "GafferSceneUI.RenderPassEditor( scriptNode )"
+
+	def __firstValidScenePlug( self, node ):
+
+		for plug in GafferScene.ScenePlug.RecursiveOutputRange( node ) :
+			if not plug.getName().startswith( "__" ):
+				return plug
+		return None
+
+	def _updateFromSet( self ) :
+
+		GafferUI.NodeSetEditor._updateFromSet( self )
+
+		# Decide what plug we're viewing.
+		self.__plug = None
+		self.__plugParentChangedConnection = None
+		self.__plugDirtiedConnections = []
+		node = self._lastAddedNode()
+		if node is not None :
+			self.__plug = self.__firstValidScenePlug( node )
+			if self.__plug is not None :
+				self.__plugParentChangedConnection = self.__plug.parentChangedSignal().connect(
+					Gaffer.WeakMethod( self.__plugParentChanged ), scoped = True
+				)
+				self.__plugDirtiedConnections.append( node.plugDirtiedSignal().connect( Gaffer.WeakMethod( self.__plugDirtied ), scoped = True ) )
+
+		self.__settingsNode["in"].setInput( self.__plug )
+
+		# call base class update - this will trigger a call to _titleFormat(),
+		# hence the need for already figuring out the plug.
+		GafferUI.NodeSetEditor._updateFromSet( self )
+
+		self.__setPathListingPath()
+
+	def _updateFromContext( self, modifiedItems ) :
+
+		if any( not i.startswith( "ui:" ) for i in modifiedItems ) :
+			self.__setPathListingPath()
+
+	def _titleFormat( self ) :
+
+		return GafferUI.NodeSetEditor._titleFormat(
+			self,
+			_maxNodes = 1 if self.__plug is not None else 0,
+			_reverseNodes = True,
+			_ellipsis = False
+		)
+
+	@GafferUI.LazyMethod()
+	def __updateColumns( self ) :
+
+		tabGroup = self.__settingsNode["tabGroup"].getValue()
+		currentSection = self.__settingsNode["section"].getValue()
+
+		sectionColumns = []
+
+		for groupKey, sections in self.__columnRegistry.items() :
+			if IECore.StringAlgo.match( tabGroup, groupKey ) :
+				section = sections.get( currentSection or None, {} )
+				sectionColumns += [ c( self.__settingsNode["in"], self.__settingsNode["editScope"] ) for c in section.values() ]
+
+		self.__pathListing.setColumns( [ self.__renderPassNameColumn, self.__renderPassActiveColumn ] + sectionColumns )
+
+	def __settingsPlugSet( self, plug ) :
+
+		if plug in ( self.__settingsNode["section"], self.__settingsNode["tabGroup"] ) :
+			self.__updateColumns()
+
+	def __plugDirtied( self, plug ) :
+
+		if self.__settingsNode["in"] is not None and plug == self.__settingsNode["in"]["globals"] :
+			self.__setPathListingPath()
```

Comment:
`self.__settingsNode["in"]` is never going to be `None` - maybe you meant `getInput() is not None`? I don't know why we need this at all though - when the globals are dirtied, the inspector columns should be signalling dirtiness so that the PathListingWidget updates automatically, shouldn't they?

Reply:
Yep, we don't need this at all. Removed in e602807